### PR TITLE
OWSM v4 Recipe

### PR DESCRIPTION
**Pull request by pyf98**
## What did you change?

This PR adds the recipe for our latest OWSM v4 models.

- Paper: https://arxiv.org/abs/2506.00338 (accepted at INTERSPEECH'25)
- Models: https://huggingface.co/collections/pyf98/open-whisper-style-speech-models-owsm-66d5312c1c9a1508189192cd

## Tasks
- [x] OWSM v4 training configs
- [x] OWSM-CTC v4 training configs
- [x] Data cleaning scripts (see our paper for detailed descriptions)

### CODE REVIEW

**Comment by sw005320** [comment]
This pull request introduces a new recipe for the `owsm_v4` dataset in the `s2t1` directory, adding configuration files, scripts, and templates for speech-to-text training and inference. The changes include backend selection for job scheduling, feature extraction configurations, training settings, and symbolic links to shared resources.

### Backend and Job Scheduling Configuration:
* [`egs2/owsm_v4/s2t1/cmd.sh`](diffhunk://#diff-ff661e2094e673b57a82b77c42d985b90eeeed98c788f36ad9012990f32b36e1R1-R110): Added backend selection logic for job scheduling (e.g., local, slurm, ssh) and unified interface for `run.pl`, `queue.pl`, `slurm.pl`, and `ssh.pl`.

### Feature Extraction Settings:
* [`egs2/owsm_v4/s2t1/conf/fbank.conf`](diffhunk://#diff-8383691d930d5fdd692acfae2ac48b8271a91a97e805f27af2fc89e65f53b926R1-R2): Configured mel-frequency filter bank with 80 mel bins and a sampling frequency of 16,000 Hz.
* [`egs2/owsm_v4/s2t1/conf/pitch.conf`](diffhunk://#diff-80ccd4072f31f3806e799fe126e67d1f17d796b8869103e90a633003d7eb68e6R1): Set the sampling frequency to 16,000 Hz for pitch extraction.

### Training and Model Configuration:
* [`egs2/owsm_v4/s2t1/conf/train_conv2d8_size1024_e18_d18_mel128.yaml`](diffhunk://#diff-2af216785ea4eb336374b67aba242b87d6f3c66f0c8b04f9a6242ff44cf7936aR1-R119): Added detailed configurations for a large model with 1024 encoder output size, 18 encoder/decoder blocks, and 128 mel features. Includes optimization and scheduler settings.
* [`egs2/owsm_v4/s2t1/conf/train_conv2d8_size384_e6_d6_mel128.yaml`](diffhunk://#diff-584f79e1d6abef5f7fd95ac65db4bc7055b9a00a3925173059679035a57ab457R1-R119): Added configurations for a smaller model with 384 encoder output size, 6 encoder/decoder blocks, and similar mel features.

### Shared Resources and Templates:
* Symbolic links created for shared resources such as `db.sh`, `path.sh`, `pyscripts`, `scripts`, `s2t.sh`, and `utils` to reuse existing templates. [[1]](diffhunk://#diff-af01fb679edcf5126c985528411ce3a1f1391a3fef50f6e6e3804ade99c3598cR1) [[2]](diffhunk://#diff-29bcfd481a9806bcaba437518c3e78776425e84fa2aafde307032cd0a548980eR1) [[3]](diffhunk://#diff-cf3f0d35e7f58b5d2aa011010ceb0c0e6b050cde8e5428428bbb3118287e6b73R1) [[4]](diffhunk://#diff-bf4fde46d89d185f4f3bbf948a475be58447caabd848c5f2c99561665c3e9ffcR1) [[5]](diffhunk://#diff-717fccb59b074bff6b3ad27678eeebe4d20d27ff4c436f61bd1f3c6b7bcf2469R1) [[6]](diffhunk://#diff-0f4d8a4106c1428c7e00238b252c5db095d28bbfaed224712d67d59062ba6827R1)

### Execution Script:
* [`egs2/owsm_v4/s2t1/run.sh`](diffhunk://#diff-7997e40b843b00b0a5df6036d8d503d618b1ad0e5a7da1bc459d1181fc494a1dR1-R34): Added a script to execute the recipe, specifying training, validation, and testing datasets, as well as configuration files and parameters for training and inference.

File: egs2/owsm_v4/s2t1/README.md
Comment (sw005320) [inline egs2/owsm_v4/s2t1/README.md:1]: Can you also add some selected results?

**Comment by sw005320** [comment]
@pyf98, any update?
Please finish it before Interspeech

File: egs2/owsm_v4/s2t1/README.md
Comment (pyf98) [reply egs2/owsm_v4/s2t1/README.md:1]: added some figures.

**Comment by pyf98** [comment]
The recipe has been uploaded.

File: egs2/owsm_v4/s2t1/local/filter_lid.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import json
+from argparse import ArgumentParser
+
+import numpy as np
+from tqdm import tqdm
+
+from utils import TO_ISO_LANGUAGE_CODE
+
+
+def norm_speech_lang(lang):
```
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/filter_lid.py:16]: It’s better to leave comments about this.
(Here and README)

File: egs2/owsm_v4/s2t1/local/filter_lid.py
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/filter_lid.py:1]: Add the description about this script and corresponding descriptions in the paper

File: egs2/owsm_v4/s2t1/local/filter_score.py
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/filter_score.py:1]: Ditto
Add the description about this script and corresponding descriptions in the paper

File: egs2/owsm_v4/s2t1/local/get_longform_from_reseg.py
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/get_longform_from_reseg.py:1]: Ditto
Add the description about this script and corresponding descriptions in the paper

File: egs2/owsm_v4/s2t1/local/data.sh
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/data.sh:1]: Please add a description and corresponding pointers in the paper

File: egs2/owsm_v4/s2t1/local/ctc_seg.py
Comment (sw005320) [inline egs2/owsm_v4/s2t1/local/ctc_seg.py:1]: Add the description about this script and corresponding descriptions in the paper

File: egs2/owsm_v4/s2t1/README.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+### English ASR
+
+![English ASR WER vs inference speed](local/en-asr.png)
+
+### Multilingual ASR
+
+![Multilingual ASR WER](local/fleurs.png)
+
+
+## Data Cleaning
```
Comment (sw005320) [inline egs2/owsm_v4/s2t1/README.md:28]: You can also record some tricks (e.g., LID normalization) here so that people can fully follow the process.

**Review by sw005320 — COMMENTED** [review]
The PR looks very good.
My comments are mostly about adding the descriptions

**Comment by sw005320** [comment]
@pyf98, please deal with CI issues, copilot reviews, and my comments.
Then, I can merge this PR.

**Comment by sw005320** [comment]
Please fix the CI issues

**Comment by sw005320** [comment]
@pyf98, let's finish this PR, which will be a good outcome together with the best student paper award.

**Comment by sw005320** [comment]
Thanks, @pyf98!

**Comment by sw005320** [comment]
You may mention that this is the award-wiining technique in README
